### PR TITLE
Implement testing best practices

### PR DESCRIPTION
**Pull request by daxmc99**
Perform some refactoring to fall in line with our testing best pratices
https://about.sourcegraph.com/handbook/engineering/languages/testing_go_code#testing-time

### CODE REVIEW

File: internal/version/version.go [hunk 1]
```diff
@@ -33,20 +33,24 @@ func Mock(mockVersion string) {
...
+}
+
 // timestamp is the build timestamp configured at build time via ldflags like this:
 // -ldflags "-X github.com/sourcegraph/sourcegraph/internal/version.timestamp=$UNIX_SECONDS"
 var timestamp = devTimestamp
 
 // HowLongOutOfDate returns a time in months since this build of Sourcegraph was created. It is
 // based on a constant baked into the Go binary at build time.
-func HowLongOutOfDate() (int, error) {
+func HowLongOutOfDate(now time.Time) (int, error) {
```
Comment (efritz) [inline internal/version/version.go:47]: If this also takes the global timestamp set by -X flags you can get rid of the MockTimestamp stuff altogether.

Comment (daxmc99) [reply internal/version/version.go:47]: True but then all callers would need to access `timestamp` which is an intentionally private variable.

Comment (efritz) [reply internal/version/version.go:47]: Ah ok I assumed it was all package local.

**Comment by unknwon** [comment]
Nice :)

**Pull request merged**